lib/src/main/java/de/edux/augmentation/effects/BrightnessAugmentation.java

```diff
@@ -0,0 +1,72 @@
+package de.edux.augmentation.effects;
+
+import de.edux.augmentation.core.AbstractAugmentation;
+
+import java.awt.*;
+import java.awt.image.BufferedImage;
+
+
+/**
+ * This augmentation technique adjusts an image's brightness value by a specified amount
+ */
+public class BrightnessAugmentation extends AbstractAugmentation {
+
+
+    private final double brightnessFactor;
+
+    /**
+     * Initialize the class with a double value between -1 and 1 indicating the brightness factor to be applied
+     * A value greater than 0 will increase brightness
+     * A value lower than 0 will decrease brightness
+     *
+     * @param brightness the brightness factor
+     * @throws IllegalArgumentException when the brightness value is not between -1 and 1
+     */
+    public BrightnessAugmentation(double brightness) throws IllegalArgumentException{
+        if (brightness < -1 || brightness > 1)
+            throw new IllegalArgumentException("Brightness value must be between -1 and 1");
+
+        this.brightnessFactor = brightness;
+    }
+
+    /**
+     * Loop through each image pixel and multiply the pixel value by the brightness value
+
+     * @param image The image to augment.
+     * @return the image with brightness augmentation applied to it
+     */
+    @Override
+    public BufferedImage apply(BufferedImage image) {
+        BufferedImage augmentedImage =
+                new BufferedImage(image.getWidth(), image.getHeight(), image.getType());
+
+        for (int y = 0; y < image.getHeight(); y++){
+            for (int x = 0; x< image.getWidth(); x++){
+                int rgb = image.getRGB(x,y);
+                Color color = new Color(rgb);
```

Comment:
Let say we work with 1000x1000 Images. 1.000.000 Pixel there. You create 1.000.000 times new object here (2x 1.000.000) in this for loop. This is a critical thing here.

Instead I suggest working with primitives here.

`          int rgb = image.getRGB(x, y);
            int alpha = (rgb >> 24) & 0xff;
            int red = (rgb >> 16) & 0xff;
            int green = (rgb >> 8) & 0xff;
            int blue = rgb & 0xff;`